use-cases/xtclim/src/anomaly.py

```diff
@@ -1,114 +1,182 @@
-        train_avg_losses, _, tot_train_losses, _ = evaluate(cvae_model, trainloader, 
-                                                        trainset, device, 
-                                                        criterion, 
-                                                        pixel_wise_criterion)
-        test_avg_losses, _, tot_test_losses, _ = evaluate(cvae_model, testloader, 
-                                                    testset, device, criterion, 
-                                                    pixel_wise_criterion)
-        for i in range(1, n_avg):
-            train_avg_loss, _, train_losses, _ = evaluate(cvae_model, trainloader, 
-                                                    trainset, device, criterion, 
-                                                    pixel_wise_criterion)
-            tot_train_losses = list(map(add, tot_train_losses, train_losses))
-            train_avg_losses += train_avg_loss
-            test_avg_loss, _, test_losses, _ = evaluate(cvae_model, testloader,
-                                                    testset, device, criterion, 
-                                                    pixel_wise_criterion)
-            tot_test_losses = list(map(add, tot_test_losses, test_losses))
-            test_avg_losses += test_avg_loss
-        tot_train_losses = np.array(tot_train_losses)/n_avg
-        tot_test_losses = np.array(tot_test_losses)/n_avg
-        train_avg_losses = train_avg_losses/n_avg
-        test_avg_losses = test_avg_losses/n_avg
-        
-        pd.DataFrame(tot_train_losses).to_csv(f"../outputs/train_losses_{season}1d_allssp.csv")
-        pd.DataFrame(tot_test_losses).to_csv(f"../outputs/test_losses_{season}1d_allssp.csv")
-        print('Train average loss:', train_avg_losses)
-        print('Test average loss:', test_avg_losses)
-
-
-if future_evaluation:
-    for season in ['winter_', 'spring_', 'summer_', 'autumn_']:
-        
-        # load previously trained model
-        cvae_model = model.ConvVAE().to(device)
-        cvae_model.load_state_dict(torch.load(f'../outputs/cvae_model_{season}1d.pth'))
-
-        for scenario in ['585', '370', '245', '126']:
-            
-            # projection set and data loader
-            proj_time = pd.read_csv(f"../input/dates_proj_{season}data.csv")
-            proj_data = np.load(f"../input/preprocessed_1d_proj{scenario}_{season}data_allssp.npy")
-            n_proj = len(proj_data)
-            projset = [ ( torch.from_numpy(np.reshape(proj_data[i], (3, 32, 32))), 
-                            proj_time['0'][i] ) for i in range(n_proj) ]
-            projloader = DataLoader(
-                projset, batch_size=1, shuffle=False
+        self.evaluation = evaluation
+        self.batch_size = batch_size
+        self.seasons = seasons
+        self.config = TrainingConfiguration(
+            batch_size=batch_size,
+        )
+
+    @monitor_exec
+    def execute(self):
```

Comment:
I am a bit lost here. This method is really large and it does many different things. I would suggest to reuse the native `execute`  provided by the `TorchTrainer` to inherit all the boilerplate code to setup the strategy and loggers. Then, you can break the use case code in different methods, depending on the value of `self.evaluation`. Example:

```python
def train(self, ...):
  if self.evaluation == "past":
    self.past_evaluation(...)
  elif self.evaluation == "future":
    self.future_evaluation(...)
  else:
    raise ValueError(...)
```